addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightDiscover.java

```diff
@@ -77,61 +86,59 @@ public MilightDiscover(DiscoverResult discoverResult, int resendTimeoutInMillis,
     }
 
     public void dispose() {
+        stopResend();
         willbeclosed = true;
         datagramSocket.close();
         try {
             join(500);
```

Comment:
I added comments to stopResend(). The method does something different then you thought it does, but it was not documented at all.